src/EA.Iws.Domain/ImportNotificationAssessment/Transactions/ImportPaymentTransaction.cs

```diff
@@ -26,21 +27,39 @@ public class ImportPaymentTransaction
         public async Task Save(Guid notificationId, DateTime date, decimal amount, PaymentMethod paymentMethod,
             string receiptNumber, string comments)
         {
-            if (await transactionCalculator.PaymentIsNowFullyReceived(notificationId, amount))
+            var transaction = ImportNotificationTransaction.PaymentRecord(notificationId, date, amount,
+                paymentMethod, receiptNumber, comments);
+            var balance = await transactionCalculator.Balance(transaction.NotificationId)
+               - transaction.Credit.GetValueOrDefault()
+               + transaction.Debit.GetValueOrDefault();
+
+            transactionRepository.Add(transaction);
+
+            if (balance <= 0)
             {
                 var assessment = await assessmentRepository.GetByNotification(notificationId);
+                var transactions = await transactionRepository.GetTransactions(transaction.NotificationId);
+                transactions = transactions.Where(t => t.Credit > 0).OrderByDescending(t => t.Date).ToList();
 
-                if (assessment.Status == ImportNotificationStatus.AwaitingPayment)
+                foreach (var tran in transactions)
                 {
-                    assessment.PaymentComplete(date);
-                }
-                else if (!assessment.Dates.PaymentReceivedDate.HasValue)
-                {
-                    assessment.Dates.PaymentReceivedDate = date;
+                    if (balance == 0)
+                    {
+                        if (assessment.Status == ImportNotificationStatus.AwaitingPayment)
+                        {
+                            assessment.PaymentComplete(tran.Date);
+                        }
+                        else
+                        {
+                            assessment.Dates.PaymentReceivedDate = tran.Date;
+                        }
+
+                        break;
```

Comment:
Personally I do not like break statements, a while loop would be better.

Reply:
Agreed. Only reason why I did it this was because it's looping through a list which I prefer using a for loop. I'll update it to a while loop.

Reply:
Updated now, got rid of the break.